src/shared/Core/Trace2.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Serialization;
+
+namespace GitCredentialManager;
+
+// The different event types tracked in the TRACE2 tracing
+// system.
+public enum Event {}
+
+/// <summary>
+/// Represents the application's TRACE2 tracing system, which
+/// writes to a collector service that stores the trace information.
+/// </summary>
+public interface ITrace2 : IDisposable
+{
+    /// <summary>
+    /// Add a listener to the trace writer.
+    /// </summary>
+    /// <param name="listener">The listener to add.</param>
+    void AddListener(CollectorWriter listener);
```

Comment:
Ok, as of this commit, `ITrace2` objects _emit_ events, but do not _accept_ events. I suppose that makes sense for the purposes of start/stop events. I expect you'll update it when adding other kinds of logging.

Reply:
Or are we using "listener" in the context of a background queue thread that is receiving trace messages from the foreground thread and that it will relay trace events to the daemon (aka collector service) ?

Reply:
Yes, the `CollectorWriter` that is added here sets up the queue that receives TRACE2 messages and eventually writes them to the daemon. Perhaps `AddWriter` would be a more appropriate name...